gtn/functions/compose.cpp

```diff
@@ -450,3 +456,936 @@ Graph compose(
 
 } // namespace detail
 } // namespace gtn
+
+namespace gtn {
+namespace detail {
+namespace dataparallel {
+// Change AOS to SOA
+struct GraphDataParallel {
+  // True if a node is accept or start, false otherwise
+  std::vector<bool> accept;
+  std::vector<bool> start;
+
+  // One value per node - i-th value corresponds to i-th node
+  // Last element is the total number of arcs, so that
+  // each element and its neighbor forms a range
```

Comment:
Just checking: it looks like we aren't actually storing the last element (e.g. the total # arcs) explicitly in `inArcOffset` and `outArcOffset`?

Reply:
Yeah since we can get it from total number of arcs - I have actually gone back and forth on this - I used to have it - and then took it away - mainly because it broke the 1:1 correspondence to nodes - and looked "ugly" to me :)